File: spring-web/src/main/java/com/example/springwebex/exception/ecode/ErrorCode.java
package com.example.springwebex.exception.ecode;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum ErrorCode {
// @formatter:off
    SUCCESS(0, "success"),
    UNKNOWN(1, "unknown error");
// @formatter:on

    private final int code;
    private final String defaultMessage;

    ErrorCode(final int code, final String defaultMessage) {
        this.code = code;
        this.defaultMessage = defaultMessage;
    }

    @JsonValue
    public int getCode() {
        return this.code;
    }

    public String getDefaultMessage() {
        return this.defaultMessage;
    }

    @JsonCreator
    public static ErrorCode valueOf(final int code) {
        for (final ErrorCode ec : ErrorCode.values()) {
            if (ec.code == code) {
                return ec;
            }
        }
        throw new IllegalArgumentException();
    }
}

File: spring-web-kotlin/src/main/kotlin/com/example/springwebex/exception/ecode/ErrorCode.kt
package com.example.springwebex.exception.ecode

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class ErrorCode(
    val code: String,
    val defaultMessage: String
) {
    SUCCESS("0", "Success"),
    INTERNAL_SERVER_ERROR("500", "Internal Server Error"),
    BAD_REQUEST("400", "Bad Request"),
    UNAUTHORIZED("401", "Unauthorized"),
    FORBIDDEN("403", "Forbidden"),
    NOT_FOUND("404", "Not Found"),
    CONFLICT("409", "Conflict"),
    UNSUPPORTED_MEDIA_TYPE("415", "Unsupported Media Type"),
    VALIDATION_ERROR("422", "Validation Error"),
    UNKNOWN("999", "Unknown Error");

    @JsonValue
    fun getValue(): String = code

    companion object {
        @JsonCreator
        @JvmStatic
        fun fromValue(value: String): ErrorCode {
            return entries.find { it.code == value } ?: UNKNOWN
        }
    }
}
